File: app/src/main/java/com/github/oryanmat/trellowidget/util/PrefUtil.java
package com.github.oryanmat.trellowidget.util;

import android.content.Context;
import android.support.annotation.ColorInt;
import android.support.annotation.StringRes;

import com.github.oryanmat.trellowidget.R;

import static android.preference.PreferenceManager.getDefaultSharedPreferences;

public class PrefUtil {

    public static float getPrefTextScale(Context context) {
        return Float.parseFloat(getDefaultSharedPreferences(context).getString(
                context.getString(R.string.pref_text_size_key),
                context.getString(R.string.pref_text_size_default)));
    }

    public static int getInterval(Context context) {
        return Integer.parseInt(getDefaultSharedPreferences(context)
                .getString(context.getString(R.string.pref_update_interval_key),
                        context.getString(R.string.pref_update_interval_default)));
    }

    public static @ColorInt int getCardBackgroundColor(Context context) {
        return getColor(context, context.getString(R.string.pref_back_color_key),
                context.getResources().getInteger(R.integer.pref_back_color_default));
    }

    public static @ColorInt int getCardForegroundColor(Context context) {
        return getColor(context, context.getString(R.string.pref_fore_color_key),
                context.getResources().getInteger(R.integer.pref_fore_color_default));
    }
    public static @ColorInt int getTitleBackgroundColor(Context context) {
        if (isTitleUniqueColor(context)) {
            return getColor(context, context.getString(R.string.pref_title_back_color_key),
                    context.getResources().getInteger(R.integer.pref_title_back_color_default));
        }
        return getCardBackgroundColor(context);
    }

    public static @ColorInt int getTitleForegroundColor(Context context) {
        if (isTitleUniqueColor(context)) {
            return getColor(context, context.getString(R.string.pref_title_fore_color_key),
                    context.getResources().getInteger(R.integer.pref_title_fore_color_default));
        }
        return getCardForegroundColor(context);
    }

    public static @ColorInt int getColor(Context context, String key, int defValue) {
        return getDefaultSharedPreferences(context).getInt(key, defValue);
    }

    public static boolean isTitleUniqueColor(Context context) {
        return isEnabled(context, R.string.pref_title_use_unique_color_key);
    }

    public static boolean isTwoLineTitle(Context context) {
        return isEnabled(context, R.string.pref_two_line_title_key);
    }

    public static boolean isTitleEnabled(Context context) {
        return isEnabled(context, R.string.pref_title_onclick_key);
    }

    public static boolean isEnabled(Context context, @StringRes int key) {
        return getDefaultSharedPreferences(context).getBoolean(
                context.getString(key),
                true); // This assumes the default preferences have been loaded
    }
}

File: app/src/main/kotlin/com/github/oryanmat/trellowidget/util/PrefUtil.kt
package com.github.oryanmat.trellowidget.util

import android.content.Context
import android.preference.PreferenceManager.getDefaultSharedPreferences
import android.support.annotation.ColorInt
import android.support.annotation.StringRes
import com.github.oryanmat.trellowidget.R

internal fun Context.getPrefTextScale() =
        java.lang.Float.parseFloat(sharedPreferences().getString(
                getString(R.string.pref_text_size_key),
                getString(R.string.pref_text_size_default)))

internal fun Context.getInterval() =
        Integer.parseInt(sharedPreferences().getString(
                getString(R.string.pref_update_interval_key),
                getString(R.string.pref_update_interval_default)))

internal @ColorInt fun Context.getCardBackgroundColor() = getColor(
        getString(R.string.pref_back_color_key),
        resources.getInteger(R.integer.pref_back_color_default))

internal @ColorInt fun Context.getCardForegroundColor() = getColor(
        getString(R.string.pref_fore_color_key),
        resources.getInteger(R.integer.pref_fore_color_default))

internal fun Context.isNoTitle() =
        isEnabled(R.string.pref_no_title_key)

internal fun Context.isTwoLineTitle() =
        isEnabled(R.string.pref_two_line_title_key)

internal fun Context.isTitleUniqueColor() =
        isEnabled(R.string.pref_title_use_unique_color_key)

internal fun Context.isTitleEnabled() =
        isEnabled(R.string.pref_title_onclick_key)

internal fun Context.isEnabled(@StringRes key: Int) =
        sharedPreferences().getBoolean(getString(key), true)

internal @ColorInt fun Context.getTitleBackgroundColor(): Int = when {
    isTitleUniqueColor() -> getColor(
            getString(R.string.pref_title_back_color_key),
            resources.getInteger(R.integer.pref_title_back_color_default))
    else -> getCardBackgroundColor()
}

internal @ColorInt fun Context.getTitleForegroundColor(): Int = when {
    isTitleUniqueColor() -> getColor(
            getString(R.string.pref_title_fore_color_key),
            resources.getInteger(R.integer.pref_title_fore_color_default))
    else -> getCardForegroundColor()
}

private @ColorInt fun Context.getColor(key: String, defValue: Int) =
        sharedPreferences().getInt(key, defValue)

private fun Context.sharedPreferences() = getDefaultSharedPreferences(this)
